OMCompiler/Compiler/NFFrontEnd/NFSimplifyExp.mo

```diff
@@ -390,6 +391,18 @@ algorithm
   end match;
 end simplifyHomotopy;
 
+function simplifyDer
+  input Expression arg;
+  input Call call;
+  output Expression exp;
+algorithm
+  if Call.variability(call) < Variability.DISCRETE then
```

Comment:
Well, I guess this is enough to solve the bug.
But mathematically we could stretch this all the way to `Variability.CONTINUOUS`.

Reply:
The specification actually doesn't mention discrete variables, only continuous, constant and parameter. Dymola does not seem to allow `der` of discrete variables, so perhaps we should just forbid that? Any opinion on that @casella?

Reply:
What about implicitly discrete expressions? :slightly_smiling_face:

Reply:
> What about implicitly discrete expressions? slightly_smiling_face

Can `Real` variables be implicitly discrete?

Reply:
I guess so
https://github.com/OpenModelica/OpenModelica/blob/96891b8f7d39ed1abd12ab30db520f8057e6d86a/OMCompiler/Compiler/NFFrontEnd/NFInst.mo#L3458-L3459

Reply:
> I guess so
> 
> https://github.com/OpenModelica/OpenModelica/blob/96891b8f7d39ed1abd12ab30db520f8057e6d86a/OMCompiler/Compiler/NFFrontEnd/NFInst.mo#L3458-L3459

Oh, right. Variables become discrete if they're assigned in a when-equation, so those shouldn't be allowed in `der` either. The check currently only checks for `Variability.DISCRETE` and not `Variability.IMPLICITLY_DISCRETE` right now though, so I'll have to fix that in another PR. Seems our CI is taking a break right now though, so I guess it'll have to wait until tomorrow.

Reply:
The variability check for `der` is fixed in #8895, now also implicitly discrete variables are caught.